python/ray/autoscaler/_private/kuberay/utils.py

```diff
@@ -73,3 +84,24 @@ def parse_quantity(quantity):
 
     exponent = Decimal(exponents[suffix[0]])
     return number * (base**exponent)
+
+
+def tpu_node_selectors_to_type(topology: str, accelerator: str) -> Optional[str]:
+    """Convert Kubernetes gke-tpu nodeSelectors to TPU accelerator_type
+    for a kuberay TPU worker group.
+    Args:
+        topology: value of the cloud.google.com/gke-tpu-topology Kubernetes
+            nodeSelector, describes the physical topology of the TPU podslice.
+        accelerator: value of the cloud.google.com/gke-tpu-accelerator nodeSelector,
+            the name of the TPU accelerator, e.g. tpu-v4-podslice
+    Returns:
+        A string, accelerator_type, e.g. "v4-8".
+    """
+    if topology and accelerator:
+        generation = gke_tpu_accelerator_to_generation[accelerator]
+        # Reduce e.g. "2x2x2" to 8
+        chip_dimensions = [int(chip_count) for chip_count in topology.split("x")]
+        num_chips = reduce(lambda x, y: x * y, chip_dimensions)
+        num_cores = num_chips * 2
```

Comment:
Is it always the case that a chip has two cores?

Reply:
All current TPU chips have two cores, although this could be subject to change in the future with newer versions of TPUs.